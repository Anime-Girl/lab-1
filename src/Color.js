import React from 'react';
import Field from "./Field";
import Square from "./Square";
import hexToCMYK from './hextocmyk';
import './App.css';

class Color extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: '#FFFFF',
            CMYK: '',
        }
        //this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    // handleChange(event) {
    //    this.setState({ value: event.target.value });
    // }

    handleSubmit(val) {
        this.setState({ value: val });
        this.setState({ CMYK: hexToCMYK(this.state.value) });
        //alert(this.state.CMYK);
        //document.getElementById('square').style.backgroundColor = this.state.value;
        // document.getElementById('square').innerHTML = hexToCMYK(this.state.value);
        //val.preventDefault();
    }
    render() {
        return (
            <div className="Color">
                <Field id="field" value={this.state.value} actSubmit={this.handleSubmit} ></Field>
                <Square id='square' value={this.state.value} CMYK={this.state.CMYK} ></Square>
            </div >
        )
    }
}
export default Color;
