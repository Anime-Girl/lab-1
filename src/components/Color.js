import React from 'react';
import Field from './Field';
import Square from './Square';
import hexToCMYK from '../hextocmyk';

class Color extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      CMYK: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(val) {
    this.setState(
      {
        value: val,
        CMYK: hexToCMYK(val),
      },
    );
  }

  render() {
    return (
      <div className="Color">
        <Field id="field" value={this.state.value} actSubmit={this.handleSubmit} />
        <Square id="square" value={this.state.value} CMYK={this.state.CMYK} />
      </div>
    );
  }
}
export default Color;
