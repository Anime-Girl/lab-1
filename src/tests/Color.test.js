import React, { createElement } from 'react';
import { create } from 'react-test-renderer';
import Color from '../components/Color';

// Still makes sense to see if this Component actually renderable
describe('[RENDER] Color â€“ Snapshot test', () => {
  test('Matches the snapshot', () => {
    const color = create(<Color />);
    expect(color.toJSON()).toMatchSnapshot();
  });
});
