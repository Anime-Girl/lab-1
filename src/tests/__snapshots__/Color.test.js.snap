// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Color â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Color"
>
  <div
    class="button"
    className="Field"
  >
    Color:
    <input
      id="field"
      onChange={[Function]}
      value=""
    />
    <button
      onClick={[Function]}
      type="submit"
      value="Submit"
    >
      content
    </button>
  </div>
  <div
    className="Square"
  >
    <div
      class="rectangle"
      id="square"
      style={
        Object {
          "backgroundColor": "",
        }
      }
      type="text"
      value="this.state.value"
    />
    <h1>
      
    </h1>
  </div>
</div>
`;
